### ISSUE TITLE
Unable to prepare Maven home for project ...

### ISSUE THREAD

**Issue by romainreuillon** [digimead/sbt-osgi-manager#5]
I am trying to depend on some lib in an eclipse update repo. My build looks mostly like that:

```
  lazy val commonsSettings =
    super.settings ++
      Seq(scalaVersion := "2.11.8")

  lazy val plugin = Project("plugin", file("plugin")) settings(commonsSettings ++ DependencyManager ++ OSGiManagerWithDebug(): _*) settings (
    resolvers in OSGiConf += typeP2("Eclipse P2 update site" at "http://eclipse.ialto.com/eclipse/updates/4.2/R-4.2.1-201209141800/"),
    resolvers in OSGiConf += typeP2("GAMA update site" at "http://vps226121.ovh.net/updates/"),
    /*libraryDependencies ++= Seq(
      "biz.aQute" % "bndlib" % "2.0.0.20130123-133441",
      //"msi.gama" % "msi.gama.headless" % "1.7.0-SNAPSHOT",
      "com.google.code.findbugs" % "jsr305" % "2.0.3" % "test"),*/
    libraryDependencies in OSGiConf += typeP2(OSGi.ECLIPSE_PLUGIN % "org.eclipse.ui" % OSGi.ANY_VERSION withSources),
    copyDepTask)
```

But when I call osgiResolve I get an error, that I have hard time to understand:

```
> osgiResolve
sbt.osgi.manager.OSGiManagerException: Unable to prepare Maven home for project gama-plugin.
	at sbt.osgi.manager.Plugin$$anonfun$1.apply(Plugin.scala:158)
	at sbt.osgi.manager.Plugin$$anonfun$1.apply(Plugin.scala:149)
	at scala.collection.TraversableLike$$anonfun$map$1.apply(TraversableLike.scala:244)
	at scala.collection.TraversableLike$$anonfun$map$1.apply(TraversableLike.scala:244)
	at scala.collection.Iterator$class.foreach(Iterator.scala:727)
	at scala.collection.AbstractIterator.foreach(Iterator.scala:1157)
	at scala.collection.MapLike$DefaultKeySet.foreach(MapLike.scala:174)
	at scala.collection.TraversableLike$class.map(TraversableLike.scala:244)
	at scala.collection.AbstractSet.scala$collection$SetLike$$super$map(Set.scala:47)
	at scala.collection.SetLike$class.map(SetLike.scala:93)
	at scala.collection.AbstractSet.map(Set.scala:47)
	at sbt.osgi.manager.Plugin$.osgiResolveCommand(Plugin.scala:149)
	at sbt.osgi.manager.Plugin$$anonfun$defaultSettings$6$$anonfun$apply$1.apply(Plugin.scala:62)
	at sbt.osgi.manager.Plugin$$anonfun$defaultSettings$6$$anonfun$apply$1.apply(Plugin.scala:62)
	at sbt.Command$$anonfun$command$1$$anonfun$apply$1.apply(Command.scala:30)
	at sbt.Command$$anonfun$command$1$$anonfun$apply$1.apply(Command.scala:30)
	at sbt.Command$.process(Command.scala:93)
	at sbt.MainLoop$$anonfun$1$$anonfun$apply$1.apply(MainLoop.scala:96)
	at sbt.MainLoop$$anonfun$1$$anonfun$apply$1.apply(MainLoop.scala:96)
	at sbt.State$$anon$1.process(State.scala:184)
	at sbt.MainLoop$$anonfun$1.apply(MainLoop.scala:96)
	at sbt.MainLoop$$anonfun$1.apply(MainLoop.scala:96)
	at sbt.ErrorHandling$.wideConvert(ErrorHandling.scala:17)
	at sbt.MainLoop$.next(MainLoop.scala:96)
	at sbt.MainLoop$.run(MainLoop.scala:89)
	at sbt.MainLoop$$anonfun$runWithNewLog$1.apply(MainLoop.scala:68)
	at sbt.MainLoop$$anonfun$runWithNewLog$1.apply(MainLoop.scala:63)
	at sbt.Using.apply(Using.scala:24)
	at sbt.MainLoop$.runWithNewLog(MainLoop.scala:63)
	at sbt.MainLoop$.runAndClearLast(MainLoop.scala:46)
	at sbt.MainLoop$.runLoggedLoop(MainLoop.scala:30)
	at sbt.MainLoop$.runLogged(MainLoop.scala:22)
	at sbt.StandardMain$.runManaged(Main.scala:54)
	at sbt.xMain.run(Main.scala:29)
	at xsbt.boot.Launch$$anonfun$run$1.apply(Launch.scala:109)
	at xsbt.boot.Launch$.withContextLoader(Launch.scala:128)
	at xsbt.boot.Launch$.run(Launch.scala:109)
	at xsbt.boot.Launch$$anonfun$apply$1.apply(Launch.scala:35)
	at xsbt.boot.Launch$.launch(Launch.scala:117)
	at xsbt.boot.Launch$.apply(Launch.scala:18)
	at xsbt.boot.Boot$.runImpl(Boot.scala:41)
	at xsbt.boot.Boot$.main(Boot.scala:17)
	at xsbt.boot.Boot.main(Boot.scala)
```

Would you have any idea on what's going on ?

Cheers,
Romain

**Comment by reyman** [comment]
There is probably a problem in

**Comment by ezh** [comment]
From Plugin.scala:158
It looks like (osgiMavenPrepareHome in Compile in OSGiConf) is undefined in project gama-plugin, so we are unable to execute this task.

output from the test project:

> inspect osgi:osgiPrepareMavenHome
[info] Task: java.io.File
[info] Description:
[info]  Prepare Maven home directory
[info] Provided by:
[info]  {file:/home/ezh/remote/}remote/osgi:osgiPrepareMavenHome
[info] Defined at:
[info]  /home/ezh/remote/build.sbt:7
[info] Dependencies:
[info]  osgi:state
[info]  osgi:osgiPrepareMavenHome::streams
[info]  osgi:thisProjectRef
[info] Delegates:
[info]  osgi:osgiPrepareMavenHome
[info]  *:osgiPrepareMavenHome
[info]  {.}/osgi:osgiPrepareMavenHome
[info]  {.}/*:osgiPrepareMavenHome
[info]  */osgi:osgiPrepareMavenHome
[info]  */*:osgiPrepareMavenHome
> osgi>
{invalid input}   
> osgi:osgiP
osgiPluginInfo         osgiPrepareBndHome     osgiPrepareMavenHome   
> osgi:osgiPrepareMavenHome
[debug] [OSGi manager:Simple] Prepare Maven home directory.
[success] Total time: 0 s, completed 19.05.2016 13:59:33

Please, show the output of your project.

**Comment by reyman** [comment]
Hi @ezh , here the result : 
```

> inspect osgi:osgiPrepareMavenHome
[info] Task: java.io.File
[info] Description:
[info]  Prepare Maven home directory
[info] Provided by:
[info]  {file:/home/reyman/Projets/gama-plugin-codingcamp2016/}plugin/osgi:osgiPrepareMavenHome
[info] Defined at:
[info]  (sbt.osgi.manager.Plugin) Plugin.scala:57
[info] Dependencies:
[info]  plugin/osgi:state
[info]  plugin/osgi:osgiPrepareMavenHome::streams
[info]  plugin/osgi:thisProjectRef
[info] Delegates:
[info]  plugin/osgi:osgiPrepareMavenHome
[info]  plugin/*:osgiPrepareMavenHome
[info]  {.}/osgi:osgiPrepareMavenHome
[info]  {.}/*:osgiPrepareMavenHome
[info]  */osgi:osgiPrepareMavenHome
[info]  */*:osgiPrepareMavenHome
```

```
> osgi:osgiPluginInfo
[info] [OSGi manager:plugin] Name: sbt-osgi-manager
[info] [OSGi manager:plugin] Version: 0.3.0.26-SNAPSHOT
[info] [OSGi manager:plugin] Build: Mon May 16 02:42:20 ICT 2016 (1463341340)

```

```
> osgi:osgiPrepareMavenHome
[debug] [OSGi manager:plugin] Prepare Maven home directory.
[success] Total time: 0 s, completed 19 mai 2016 18:08:52
```

**Comment by reyman** [comment]
@ezh Hum, we finally switch to build.sbt definition of "DependencyManager". 
We don't know how to define this in `/plugin/build.scala` :/

**Comment by ezh** [comment]
Thank you for the link. I clone your project and reproduce the error. Very interesting. I hope that I'll have enough luck and time to fix it. I'll leave comment here with my progress anyway.

val taskMavenPrepareHomeKey = osgiMavenPrepareHome in Compile in OSGiConf
      EvaluateTask(extracted.structure, taskMavenPrepareHomeKey, actualState, projectRef)
is broken for your case. Sorry.

**Comment by ezh** [comment]
I add your case to test suite and fix OSGiManagerException. I'll push the fix to github 21.05 or 22.05 if everything will fine.

**Comment by ezh** [comment]
@reyman [your test case](https://github.com/digimead/sbt-osgi-manager/tree/master/src/sbt-test/osgi-manager/subproject2/project)
